ignite3/src/main/java/site/ycsb/db/ignite3/IgniteStreamerClient.java

```diff
@@ -130,28 +115,10 @@ public Status delete(String table, String key) {
   /** {@inheritDoc} */
   @Override
   public void cleanup() throws DBException {
-    synchronized (IgniteStreamerClient.class) {
-      int curInitCount = INIT_COUNT.decrementAndGet();
+    rvPublisher.close();
 
-      if (curInitCount <= 0) {
-        try {
-          rvPublisher.close();
+    rvStreamerFut.join();
 
-          rvStreamerFut.join();
-
```

Comment:
Need to perform these actions for each thread instance, not only for the last one.